server/src/main/java/com/cloud/deploy/DeploymentPlanningManagerImpl.java

```diff
@@ -1558,7 +1558,8 @@ protected boolean hostCanAccessSPool(Host host, StoragePool pool) {
 
             DiskOfferingVO diskOffering = _diskOfferingDao.findById(toBeCreated.getDiskOfferingId());
 
-            if (vmProfile.getTemplate().getFormat() == Storage.ImageFormat.ISO && vmProfile.getServiceOffering().getTagsArray().length != 0) {
+            if ((vmProfile.getTemplate().getFormat() == Storage.ImageFormat.ISO || toBeCreated.getVolumeType() == Volume.Type.ROOT)
+                    && vmProfile.getServiceOffering().getTagsArray().length != 0) {
```

Comment:
For ROOT volume from template, the disk / service offering storage tags are the same. Is it required to explicitly set the disk offering tags?

Reply:
@sureshanaparti 
yes, you are right. it is not required. this is why I converted this PR to draft today.

I thought I have reproduced the issue two days before. but today I am not able to reproduce.
I am still investigating the issue #4406 
When I found some clue I will update this PR

Reply:
@sureshanaparti @davidjumani 
found two scenarios that root disk offering is not same as vm service offering.
there might be more scenarios (for example, attach data disk as root disk ?).

please review

Reply:
@weizhouapache first scenario - when VM is deployed from ISO, what is the other scenario ?

Does attach data disk as root disk supported?

Reply:
@sureshanaparti 
as I said in the description

```
In some cases, the disk offering of ROOT disk is not same as service offering of VM.
for example
(1) create a vm but not start it, then scale vm to another service offering
(2) reinstall vm, then scale vm to another service offering

These two issues are fixed by this PR.
The issue might exist in some other scenarios, so it would be better to always update ROOT disk offering tags to the tags of vm service offering in DeploymentPlanningManagerImpl.java.
```

Reply:
@weizhouapache in both these scenarios, does the ROOT disk offering in volumes table updated with the new service offering?

Reply:
@sureshanaparti 
only the root volumes in Ready state are updated (root volumes are Allocated in these two scenarios).
the issues are fixed in this PR.